examples/DemoFile.Example.WebAssembly/Program.cs

```diff
@@ -0,0 +1,118 @@
+﻿using System.Diagnostics;
+using System.Runtime.InteropServices.JavaScript;
+using System.Text;
+
+namespace DemoFile.Example.WebAssembly;
+
+public partial class Program
+{
+    public static async Task Main(string[] args)
+    {
+        Console.WriteLine("Main()");
+    }
+
+    [JSExport]
+    public static async Task ParseToEndFromUrl(string url)
+    {
+        Console.WriteLine($"Parsing demo from url: {url}");
+        using var client = new HttpClient();
+        using var stream = await client.GetStreamAsync(url);
+        await ParseToEndFromStream(stream);
+    }
+
+    [JSExport]
+    public static async Task ParseToEndFromArray(byte[] buffer)
+    {
+        using var stream = new MemoryStream(buffer);
+        await ParseToEndFromStream(stream);
+    }
+
+    private static async Task ParseToEndFromStream(Stream stream)
+    {
+        Console.WriteLine($"Parsing started, stream length {stream.Length}");
+
+        SetDemoParseResult(string.Empty);
+        SetDemoParseProgress(0);
+
+        var demo = new DemoParser();
+        var sb = new StringBuilder(1024);
```

Comment:
This isn't exactly what I was thinking.

Rather than building up a big string of all of the events, can a message be pushed to the web page per event? Then the web page adds the string to its DOM.

e.g. RoundStart would look like:

```c#
demo.Source1GameEvents.RoundStart += e =>
{
    onUpdate($"Round start [{roundNum}]");
};
```

Where the string is passed to the web page

Reply:
Sure, it can be done.
But that's what is effectively being done right now. If you see the line below :
`if (sb.Length > 0)
                    AppendDemoParseResult(sb.ToString());`.
The string is sent only when there is something new in StringBuilder.
But you are right, it looks better if it's sent "event-based".